### DISCUSSION TITLE
Unable to recieve HLS stream of audio only AAC stream. ERR: (the stream doesn't contain an H264 track or an AAC track)

### DISCUSSION THREAD

**Discussion by raoufahmed248** [bluenviron/mediamtx#841] [Questions] [answered]
Hi! I wasn't sure if I should put this under issues since it's more than likely just a user error on my part. 
I'm trying to convert a audio only rtsp stream to a web viewable HLS stream. The issue is when I try to view the HLS stream on the browser (using: http://localhost:8888/mystream/index.m3u8 into hls.js), I get an error from the rtsp-simple-server application: 
"INF [HLS] [muxer mystream] closed (the stream doesn't contain an H264 track or an AAC track)" 

I am sending (what I'm thinking is) a AAC stream via gstreamer using this pipeline:
 gst-launch-1.0 audiotestsrc ! audioconvert ! audioresample  ! avenc_aac ! rtspclientsink location=rtsp://myuser:mypass@127.0.0.1:8554/mystream

I am able to verify that this works, by listening it to it using VLC:
vlc rtsp://localhost:8554/mystream

The error given seems to state that I'm not providing an AAC stream, however I think I'm doing the right steps to encode the audio stream as an AAC stream (using avenc_aac encoding)

Thank you for your time!

**Comment by aler9** [comment, marked as answer]
Hello, it seems that Gstreamer encodes AAC by using the [RFC 3016 standard](https://www.ietf.org/rfc/rfc3016.txt), while all other clients encodes AAC with the [RFC 3640 standard](https://datatracker.ietf.org/doc/html/rfc3640). The server supports the latter, and RFC 3016-encoded tracks are treated as generic tracks.

This may change in the future, but for the moment, you have to choose another RTSP client. For instance, gortsplib allows you to publish audio from a Gstreamer pipeline to a RTSP server:

https://github.com/aler9/gortsplib/blob/main/examples/client-publish-aac/main.go

[a comment by an automated account is not shown]
